aesara/link/numba/dispatch/basic.py

```diff
@@ -515,6 +514,47 @@ def numba_funcify_IncSubtensor(op, node, **kwargs):
     return numba_njit(incsubtensor_fn)
 
 
+@numba_funcify.register(AdvancedIncSubtensor1)
+def numba_funcify_AdvancdIncSubtensor1(op, node, **kwargs):
+    inplace = op.inplace
+    set_instead_of_inc = op.set_instead_of_inc
+    boundscheck = op.boundscheck
```

Comment:
Can't numba optimize away the check when the inputs are constant?

Reply:
Unfortunately not, because at compile time it doesn't know the shape of the array it is indexing into yet. But even if it did, I don't think the optimization model of llvm would allow an optimization like this, because it would change observable behaviour: With the optimization *no* element of the array is changed when there is an out of bounds access at the end, but without the optimization we still change the array up to this point. So unless it can prove that this can't be observed it is not allowed to move the check to the front of the loop.
Anyway, I checked it definitely doesn't do this right now.

Reply:
I guess you referred to numba, not llvm. So the C-rules don't really apply, it still doesn't do this either however...

Reply:
LLVM can and does perform constant propagation.